src/Libraries/Auth/WebAuth.php

```diff
@@ -89,12 +89,24 @@ public function signin($mailer, $username, $password, $remember = false)
             $this->setRememberToken($user);
         }
 
-        if (config()->get('two_step_verification')) {
+        if (filter_var(config()->get('2SV'), FILTER_VALIDATE_BOOLEAN)) {
 
-            $user = $this->towStepVerification($mailer, $user);
-        }
+            $otp_token = $this->generateOtpToken($user[$this->keys['usernameKey']]);
 
-        session()->set($this->authUserKey, $this->filterFields($user));
+            $time = new \DateTime();
+
+            $time->add(new \DateInterval('PT' . config()->get('otp_expiry_time') . 'M'));
+
+            $otp_expiry_time = $time->format('Y-m-d H:i');
+
+            $this->towStepVerification($mailer, $user, $otp_expiry_time, $otp_token);
+
+            return $otp_token;
```

Comment:
The expression return $otp_token returns the type string which is incompatible with the documented return type boolean.